frappe/model/base_document.py

```diff
@@ -101,13 +102,11 @@ def update(self, d):
 				"balance": 42000
 			})
 		"""
-		if "doctype" in d:
-			self.set("doctype", d.get("doctype"))
-
 		# first set default field values of base document
 		for key in default_fields:
-			if key in d:
-				self.set(key, d.get(key))
+			_val = d.get(key)
+			if _val:
+				self.set(key, _val)
```

Comment:
This isn't the same as earlier logic, this only works if `_val` is truthy. But maybe may this loop isn't required anymore?

Reply:
Shoot, you're right! What do you suggest instead of loop?

Reply:
These values are already being set in the loop below, so I think we need to understand why (if at all) do we need to set the default fields first.

Reply:
Yes, that's true. The idea is to set specific order of fields. Should we keep it? 🤔

> This isn't the same as earlier logic, this only works if _val is truthy.

I added new commit that fixes this. Can you check it out?